packages/client/lib/miner/miner.ts

```diff
@@ -286,11 +287,13 @@ export class Miner {
     )
     let index = 0
     let blockFull = false
+    const receipts: TxReceipt[] = []
     while (index < txs.length && !blockFull && !interrupt) {
       try {
-        await blockBuilder.addTransaction(txs[index], {
+        const txResult = await blockBuilder.addTransaction(txs[index], {
           skipHardForkValidation: this.skipHardForkValidation,
         })
+        this.config.saveReceipts && receipts.push(txResult.receipt)
```

Comment:
Short nit: won't make it a blocker but I think it is better to write these kind of things "in the old fashioned way" with `if (...) { // push the receipts }`, this is otherwise for a lot of developers unnecessarily hard to read (on a very very rough estimate I would assume that 50% of people know this and 50% do not or are at least not so sure or getting wonky on these kind of code parts).

Reply:
Update: I needed to do a small update to fix client test runs anyhow, so I have updated to easier to read conditionals along.

Reply:
I didn't realize that the `&&`  was such a hip new thing! :stuck_out_tongue: 

I can avoid this in the future

Reply:
The hip new programmers for sure know this, but I e.g. already needed two additional brain cycles to get this deciphered. 😂